codecs/jxl/enc/jxl_enc.cpp

```diff
@@ -31,10 +32,11 @@ val encode(std::string image, int width, int height, JXLOptions options) {
 
   uint8_t* inBuffer = (uint8_t*)image.c_str();
 
-  auto result =
-      main->SetFromSRGB(width, height, /*is_gray*/ false, /*has_alpha*/ true,
-                        /*alpha_is_premultiplied*/ true, /*pixels*/ (uint8_t*)image.c_str(),
-                        /* end */ (uint8_t*)image.c_str() + width * height * 4);
+  auto result = jxl::ConvertImage(
```

Comment:
I see this new API accepts `bits_per_alpha` and `bits_per_sample` via params, does this mean we don't need the `io.metadata.Set*` lines above anymore?

Also, a more general question (not in scope of this PR) - should we generally switch to and use C API instead? (I'm assuming that one is public given that it's the only one exposed from `include/`)

Reply:
bits_per_sample is not needed, but AlphaBits is.

This is an internal API, but AFAIU there's no encoder C API yet - it's better to switch once it is.

Reply:
> but AlphaBits is

Interesting. What's the difference between `bits_per_alpha` param and `io.metadata.SetAlphaBits`?

> This is an internal API, but AFAIU there's no encoder C API yet - it's better to switch once it is.

Fair enough, thanks for the confirmation.

Reply:
I don't think there's a difference per se, but I think we access the field before the API below sets it.